packages/worker/src/worker-options.ts

```diff
@@ -193,12 +203,72 @@ export interface WorkerOptions {
   maxTaskQueueActivitiesPerSecond?: number;
 
   /**
-   * Maximum number of Workflow tasks to execute concurrently.
-   * Adjust this to improve Worker resource consumption.
-   * @default 100
+   * Maximum number of Workflow Tasks to execute concurrently.
+   *
+   * In general, a Workflow Worker's performance is mostly network bound (due to communication latency with the
+   * Temporal server). Accepting multiple Workflow Tasks concurrently helps compensate for network latency, until the
+   * point where the Worker gets CPU bound.
+   *
+   * Increasing this number will have no impact if Workflow Task pollers can't fill available execution slots fast
+   * enough. Therefore, when adjusting this value, you may want to similarly adjust `maxConcurrentWorkflowTaskPolls`.
+   * See {@link WorkerOptions.maxConcurrentWorkflowTaskPolls} for more information.
+   *
+   * Also, setting this value too high might cause Workflow Task timeouts due to the fact that the Worker is not able
+   * to complete processing accepted Workflow Tasks fast enough. Increasing the number of Workflow threads
+   * (see {@link WorkerOptions.workflowThreadPoolSize}) may help in that case.
+   *
+   * As a general guidelines:
+   * - High latency to Temporal Server => Increase this number
+   * - Very short Workflow Tasks (no lengthy Local Activities) => increase this number
+   * - Very long/heavy Workflow Histories => decrease this number
+   * - Low CPU usage despite backlog of Workflow Tasks => increase this number
+   * - High number of Workflow Task timeouts => decrease this number
+   *
+   * In our reference performance test against Temporal Cloud, running with a single Workflow thread and the Reuse V8
+   * Context option enabled, we reached peak performance with a `maxConcurrentWorkflowTaskExecutions` of `120`, and
+   * `maxConcurrentWorkflowTaskPolls` of `60`. Your millage may vary.
+   *
+   * Can't be lower than 2 if `maxCachedWorkflows` is non-zero.
+   * @default 40
    */
   maxConcurrentWorkflowTaskExecutions?: number;
 
+  /**
+   * Maximum number of Workflow Tasks to poll concurrently.
+   *
+   * In general, a Workflow Worker's performance is mostly network bound (due to communication latency with the
+   * Temporal server). Polling multiple Workflow Tasks concurrently helps compensate for this latency, by ensuring that
+   * the Worker is not starved waiting for the server to return new Workflow Tasks to execute.
+   *
+   * This setting is highly related with {@link WorkerOptions.maxConcurrentWorkflowTaskExecutions}. In various
+   * performance tests, we generally got optimal performance by setting this value to about half of
+   * `maxConcurrentWorkflowTaskExecutions`. Your millage may vary.
+   *
+   * Setting this value higher than needed will generally not have negative impact on this Worker's performance; your
+   * server may however impose a limit on the total number of concurrent Workflow Task pollers.
```

Comment:
But it may have a negative impact on the cluster's perf and may result in workflow tasks timing out if they're not processed in a timely fashion.

Reply:
As far as I could see in perf tests, task timeouts were linked to setting `maxConcurrentWorkflowTaskExecutions` too high, not to increasing `maxConcurrentWorkflowTaskPolls`. Core won't poll anyway if there is no execution slot available.

Still, I will change the phrasing to mention that too high might be negative on the server cluster.